<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>730</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>730</width>
    <height>600</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>730</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Robot Remote Debugger</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="gbRobotController">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>60</y>
      <width>331</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Robot Controller</string>
    </property>
    <widget class="QRadioButton" name="rbJoint">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>60</y>
       <width>89</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Joint</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="rbXyzwpr">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>89</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Xyzwpr</string>
     </property>
    </widget>
    <widget class="QLabel" name="lStep">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>80</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Step:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sbStep">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>78</y>
       <width>42</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
     <property name="value">
      <number>50</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ1pos">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>+J1/X</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ2pos">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>110</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>+J2/Y</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ3pos">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>110</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>+J3/Z</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ4pos">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>110</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>+J4/W</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ5pos">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>110</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>+J5/P</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ6pos">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>110</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>+J6/R</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ1neg">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>160</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>-J1/X</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ4neg">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>160</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>-J4/W</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ5neg">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>160</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>-J5/P</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ2neg">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>160</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>-J2/Y</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ3neg">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>160</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>-J3/Z</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbJ6neg">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>160</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>-J6/R</string>
     </property>
    </widget>
    <widget class="QLabel" name="lCoordType">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Coordinate Type:</string>
     </property>
    </widget>
   </widget>
   <widget class="QSpinBox" name="sbSpeed">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>20</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
   </widget>
   <widget class="QLabel" name="lSpeed">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>25</y>
      <width>54</width>
      <height>12</height>
     </rect>
    </property>
    <property name="text">
     <string>Speed</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="gbRDO">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>290</y>
      <width>331</width>
      <height>61</height>
     </rect>
    </property>
    <property name="title">
     <string>Robot Digital Output</string>
    </property>
    <widget class="QSpinBox" name="sbRDO">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>20</y>
       <width>42</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>8</number>
     </property>
    </widget>
    <widget class="QLabel" name="lChannel">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>24</y>
       <width>54</width>
       <height>12</height>
      </rect>
     </property>
     <property name="text">
      <string>Channel:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbRDOOn">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>20</y>
       <width>51</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>On</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbRDOOff">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>20</y>
       <width>51</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Off</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gbRobotCoord">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>10</y>
      <width>341</width>
      <height>341</height>
     </rect>
    </property>
    <property name="title">
     <string>Robot Coordinate Value</string>
    </property>
    <widget class="QLabel" name="lUT">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>20</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>UT:</string>
     </property>
    </widget>
    <widget class="QLabel" name="lUF">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>20</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>UF:</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="gbJointValue">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>321</width>
       <height>111</height>
      </rect>
     </property>
     <property name="title">
      <string>Joint Coordinate Value</string>
     </property>
     <widget class="QPlainTextEdit" name="ptJointValue">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>20</y>
        <width>261</width>
        <height>41</height>
       </rect>
      </property>
      <property name="plainText">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="pbSetJoint">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>70</y>
        <width>81</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Set Joint</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJointValu">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Joint</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="gbCartesianValue">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>321</width>
       <height>141</height>
      </rect>
     </property>
     <property name="title">
      <string>Cartesian Coordinate Value</string>
     </property>
     <widget class="QPlainTextEdit" name="ptXyzwprValue">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>20</y>
        <width>261</width>
        <height>41</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="leConfigValue">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>70</y>
        <width>261</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="lXyzwprValue">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Xyzwpr</string>
      </property>
     </widget>
     <widget class="QLabel" name="lConfigValue">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Config</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pbSetCartesian">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>100</y>
        <width>81</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Set Xyzwpr</string>
      </property>
     </widget>
    </widget>
    <widget class="QLineEdit" name="leUT">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>20</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="leUF">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>20</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gbRobotInfo">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>360</y>
      <width>691</width>
      <height>191</height>
     </rect>
    </property>
    <property name="title">
     <string>Robot Information</string>
    </property>
    <widget class="QLabel" name="lSocketInfo">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>20</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Socket Messages:</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="gbJointInfo">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>20</y>
       <width>121</width>
       <height>161</height>
      </rect>
     </property>
     <property name="title">
      <string>Joint Info</string>
     </property>
     <widget class="QLabel" name="lJ1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J1:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J2:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J3:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J4:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J5:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J6:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ6Value">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>120</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J6:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ3Value">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>60</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J3:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ2Value">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>40</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J2:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ4Value">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>80</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J4:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ5Value">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>100</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J5:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lJ1Value">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>20</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>J1:</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="gbCartesianInfo">
     <property name="geometry">
      <rect>
       <x>480</x>
       <y>20</y>
       <width>201</width>
       <height>161</height>
      </rect>
     </property>
     <property name="title">
      <string>Cartesian Info</string>
     </property>
     <widget class="QLabel" name="lRValue">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>120</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>R</string>
      </property>
     </widget>
     <widget class="QLabel" name="lZValue">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>60</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Z</string>
      </property>
     </widget>
     <widget class="QLabel" name="lWValue">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>80</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>W</string>
      </property>
     </widget>
     <widget class="QLabel" name="lPValue">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>100</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>P</string>
      </property>
     </widget>
     <widget class="QLabel" name="lR">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>R:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lZ">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Z:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lY">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Y:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lW">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>W:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lXValue">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>20</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>X</string>
      </property>
     </widget>
     <widget class="QLabel" name="lYValue">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>40</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Y</string>
      </property>
     </widget>
     <widget class="QLabel" name="lP">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>P:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lX">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>X:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lConfig">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>20</y>
        <width>54</width>
        <height>12</height>
       </rect>
      </property>
      <property name="text">
       <string>Config:</string>
      </property>
     </widget>
     <widget class="QLabel" name="lConfigInfo">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>40</y>
        <width>71</width>
        <height>51</height>
       </rect>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
      </property>
     </widget>
    </widget>
    <widget class="QListWidget" name="lwMsg">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>301</width>
       <height>141</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pbGetCoord">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>20</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Get Coord</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="cbAuto">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>23</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Auto </string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>730</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSetting">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionIP_Config"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
   </widget>
   <addaction name="menuSetting"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionIP_Config">
   <property name="text">
    <string>IP Config</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
